chaosgcp/iam/controls/policy.py

```diff
@@ -0,0 +1,304 @@
+# Copyright 2024 Google LLC
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List
+from chaoslib.exceptions import InterruptExecution
+from chaoslib.types import Configuration, Experiment, Journal, Secrets
+import google.auth
+import googleapiclient.discovery
+import datetime
+import time
+import logging
+from datetime import timezone
+
+logger = logging.getLogger("chaostoolkit")
+
+__all__ = ["before_experiment_control", "after_experiment_control"]
+
+set_current_time = datetime.datetime.now(timezone.utc)
+
+
+def before_experiment_control(
+    context: Experiment,
+    project_id: str,
+    roles: List,
+    members: List,
+    configuration: Configuration = None,
+    secrets: Secrets = None,
+    iam_propogation_sleep_time_in_minutes=2,
+    expiry_time_in_minutes=10,
```

Comment:
by convention, we always end the signature withe the configuration and secrets. Python and chaostoolkit won't care, but this make code readability more consistent.

Reply:
Also make sure to add a type for both arguments